src/Symfony/Component/Validator/Constraint.php

```diff
@@ -278,10 +278,7 @@ public function validatedBy(): string
      * Returns whether the constraint can be put onto classes, properties or
      * both.
      *
-     * This method should return one or more of the constants
-     * Constraint::CLASS_CONSTRAINT and Constraint::PROPERTY_CONSTRAINT.
-     *
-     * @return string|string[] One or more constant values
+     * @return self::CLASS_CONSTRAINT|self::PROPERTY_CONSTRAINT|array<self::CLASS_CONSTRAINT|self::PROPERTY_CONSTRAINT> One or more constant values
```

Comment:
wouldn't this work?

```suggestion
     * @return self::*_CONSTRAINT|array<self::*_CONSTRAINT> One or more constant values
```

Reply:
It works: https://phpstan.org/r/554d2fef-1a63-4b13-880a-7afe3ad0a8c0

I hesitated, because we could unintentionally include new constants ending with `_CONSTRAINT`. But if it seems OK to you, it works for me 👍

Reply:
It appears to not work for Psalm (https://psalm.dev/r/25734b245c), which is a requirement for us (as this is what we use in our own CI).

Reply:
Oh right, let's keep it as-is then. Thanks!

Reply:
then:
```suggestion
     * @return self::CLASS_CONSTRAINT|self::PROPERTY_CONSTRAINT|array<self::CLASS_CONSTRAINT|self::PROPERTY_CONSTRAINT>
```

Reply:
Indeed!